src/exercises/chromium/interoperability-with-cpp.md

```diff
@@ -0,0 +1,71 @@
+# Exercise: Interoperability with C++
+
+## Part one
+
+* In the Rust file you previously created, add a `#[cxx::bridge]` which specifies a single function,
+  to be called from C++, called `hello_from_rust`, taking no parameters and returning
+  no value.
+* Modify your previous `hello_from_rust` function to remove `extern "C"` and `#[no_mangle]`.
+  This is now just a standard Rust function.
+* Modify your `gn` target to build these bindings.
+* In your C++ code, remove the forward-declaration of `hello_from_rust`. Instead, include
+  the generated header file.
+* Build and run!
+
+## Part two
+
+It's a good idea to play with cxx a little. It helps you think about how flexible
+Rust in Chromium actually is.
+
+Some things to try:
+
+* Call back into C++ from Rust. You will need:
+  * An additional header file which you can `include!` from your `cxx::bridge`.
+    You'll need to declare your C++ function in that new header file.
+  * An `unsafe` block to call such a function, or alternatively specify the `unsafe`
+    keyword in your `#[cxx::bridge]` [as described here][0].
+  * You may also need to `#include "third_party/rust/cxx/v1/crate/include/cxx.h"`
+* Pass a C++ string from C++ into Rust.
+* Pass a reference to a C++ object into Rust.
+* Intentionally get the Rust function signatures mismatched from the `#[cxx::bridge]`,
+  and get used to the errors you see.
+* Intentionally get the C++ function signatures mismatched from the `#[cxx::bridge]`,
+  and get used to the errors you see.
+* Pass a `std::unique_ptr` of some type from C++ into Rust, so that Rust
+  can own some C++ object.
+* Create a Rust object and pass it into C++, so that C++ owns it. (Hint:
+  you need a `Box`).
+* Declare some methods on a C++ type. Call them from Rust.
+* Declare some methods on a Rust type. Call them from C++.
```

Comment:
It's an already quite a long list.  But I wonder if we should (maybe just in passing) also mention:

* Error handling (e.g. having to translate `Result` into `bool` or manually monomorphize into `ResultOfFoo` or something like that)
* Slices